src/ocean/core/BitArray.d

```diff
@@ -189,13 +191,8 @@ struct BitArray
     }
     body
     {
-        size_t mDim=len/32;
-        ptr[0..mDim] = rhs.ptr[0..mDim];
-        int rest=cast(int)(len & cast(size_t)31u);
```

Comment:
ok, I see your point about using current array functionality rather than `memcpy`. Anyway your suggestion is incorrect in the way `mDim` is computed, it should be `(len+31)/32` or use `dim()` (because you are not copying the last dimension).